pysrc/bytewax/recovery.py

```diff
@@ -1,32 +1,52 @@
 """Bytewax's state recovery machinery.
 
+Preparation
+-----------
+
 This allows you to **recover** a stateful dataflow; it will let you
-resume processing and output due to system outage or error without
-re-processing all initial data to re-calculate all internal state.
+resume processing and output due to a failure without re-processing
+all initial data to re-calculate all internal state.
 
-0. Pick a **recovery store** in which internal state will be
-   continuously backed up. Different storage systems will have
-   different performance and reconfiguration trade-offs.
+0. Pick a **recovery store** in which internal state and progress will
+be continuously backed up. Different storage systems will have
+different performance and reconfiguration trade-offs.
 
-1. Create a **recovery config** for connecting to your recovery store.
+1. Create a **recovery config** for connecting to your recovery
+store. See the class types in this module for the datstores supported.
 
 2. Pass that created config as the `recovery_config` argument to the
-   entry point running your dataflow (e.g. `bytewax.run_cluster()`).
+entry point running your dataflow (e.g. `bytewax.cluster_main()`).
 
-3. Run your dataflow consuming input either from the beginning or
-   resuming input from the last finalized epoch.
+3. Run your dataflow! It will start backing up state data
+automatically.
 
 The epoch is the unit of recovery. It is your responsibility to design
-your input builders or iterators in such a way that it can be resumed
-from a given epoch; if it can't (because the input is ephemeral) then
-this feature won't allow complete recovery. It is possible that your
-output systems will see duplicate data right around the resumed epoch;
-design your systems to be idempotent or support at-least-once
+your input handlers in such a way that it jumps to the point in the
+input that corresponds to the `resume_epoch` argument; if it can't
+(because the input is ephemeral) you can still recover the dataflow,
+but the lost input is unable to be replayed so the output will be
+different.
+
+It is possible that your output systems will see duplicate data right
+around the resume epoch; design your systems to support at-least-once
 processing.
 
-Bytewax will automatically save and recover state from the recovery
-store during the execution of stateful operators. You know an operator
-is stateful when it takes a `step_id` argument.
+Recovery
+--------
+
+If the dataflow fails, first you must fix whatever underlying fault
+caused the issue. That might mean deploying new code which fixes a
+bug, removing bad data from a Kafka stream, or resolving an issue with
```

Comment:
Good point. I think I was thinking of bumping consumer group offsets to skip something, but that also isn't relevant here because we aren't using those. Will change.